### PR TITLE
Add IPV6_PKTINFO, Correct documentationn for IPV6_RECVPKTINFO

### PR DESCRIPTION
**Pull request by ishanjain28**
`IPV6_PKTINFO` was missing from `sockopt` so I added it and the documentation for `IPV6_RECVPKTINFO` incorrectly said it was enabling the flag for `IPV6_PKTINFO` so I corrected that.

### CODE REVIEW

File: src/sys/socket/sockopt.rs [hunk 1]
```diff
@@ -858,14 +866,34 @@ sockopt_impl!(
...
 );
+#[cfg(any(
+    target_os = "android",
+    target_os = "freebsd",
+    target_os = "ios",
+    target_os = "linux",
+    target_os = "macos",
+    target_os = "netbsd",
+    target_os = "openbsd",
+))]
```
Comment (SteveLauC) [inline src/sys/socket/sockopt.rs:885]: This symbol seems to be available on DragonFlyBSD and other Apple targets, if we add them, then this can be changed to:

```suggestion
#[cfg(any(linux_android, bsd))]
```

**Review by SteveLauC — CHANGES_REQUESTED** [review]
Thanks for your interest in contributing to Nix, please rebase your branch so that the CI can work (We have migrated from Cirrus to GitHub Action)

**Comment by evanrittenhouse** [comment]
Hi, I need to be able to set this sockopt as well - should I make a new PR? This one seems a bit stale cc @SteveLauC

**Comment by SteveLauC** [comment]
> Hi, I need to be able to set this sockopt as well - should I make a new PR? This one seems a bit stale cc @SteveLauC

Hi, thanks for your interest in filing a PR for this! Please go ahead:)

**Comment by ishanjain28** [comment]
hey @evanrittenhouse, Can I revive this PR instead? will rebase it today

File: src/sys/socket/sockopt.rs (hunk 1 above)
Comment (ishanjain28) [reply src/sys/socket/sockopt.rs:885]: Updated!

File: src/sys/socket/sockopt.rs [hunk 2]
```diff
@@ -905,18 +905,31 @@ sockopt_impl!(
     libc::IP_PKTINFO,
     bool
 );
-#[cfg(any(linux_android, target_os = "freebsd", apple_targets, netbsdlike))]
+#[cfg(any(linux_android))]
```
Comment (ishanjain28) [inline src/sys/socket/sockopt.rs:908]: I removed BSD target because I can't find `IPV6_RECVPKTINFO` in `ip6` bsd man pages. https://man.freebsd.org/cgi/man.cgi?query=ip6&sektion=4

**Comment by ishanjain28** [comment]
I must be missing some thing. I can not find `IPV6_RECVPKTINFO` in bsd man pages. Is it still present there and I am looking at the wrong place or do I update the tests ?

File: src/sys/socket/sockopt.rs (hunk 2 above)
Comment (SteveLauC) [reply src/sys/socket/sockopt.rs:908]: It is available on FreeBSD: https://github.com/freebsd/freebsd-src/blob/ef4f4a44d9134ddadede0e2e6e658d0688c5ab3c/sys/netinet6/in6.h#L446

Comment (SteveLauC) [reply src/sys/socket/sockopt.rs:908]: Also defined in the libc crate: https://docs.rs/libc/latest/x86_64-unknown-freebsd/libc/constant.IPV6_RECVPKTINFO.html

Comment (SteveLauC) [reply src/sys/socket/sockopt.rs:908]: The CI failure on BSDs is introduced by this change, please revert it

File: src/sys/socket/sockopt.rs [hunk 3]
```diff
@@ -905,18 +905,31 @@ sockopt_impl!(
...
 #[cfg(feature = "net")]
 sockopt_impl!(
     #[cfg_attr(docsrs, doc(cfg(feature = "net")))]
-    /// Set delivery of the `IPV6_PKTINFO` control message on incoming
+    /// Set delivery of the `IPV6_RECVPKTINFO` control message on incoming
```
Comment (SteveLauC) [inline src/sys/socket/sockopt.rs:912]: Emm, why this change? The old document seems to be correct, it asks the incoming packages to contain the `IPV6_PKTINFO` info

File: src/sys/socket/sockopt.rs [hunk 4]
```diff
@@ -905,18 +905,31 @@ sockopt_impl!(
...
     libc::IPV6_RECVPKTINFO,
     bool
 );
+
+#[cfg(any(linux_android, bsd))]
+#[cfg(feature = "net")]
+sockopt_impl!(
+    #[cfg_attr(docsrs, doc(cfg(feature = "net")))]
+    /// Set delivery of the `IPV6_PKTINFO` control message on incoming
+    /// datagrams.
```
Comment (SteveLauC) [inline src/sys/socket/sockopt.rs:926]:
```suggestion
    /// Pass an `IPV6_PKTINFO` ancillary message that contains a in6_pktinfo
    /// structure that supplies some information about the incoming packet.
```

Let's make it consistent with the IPv4 one:)

File: src/sys/socket/sockopt.rs (hunk 2 above)
Comment (SteveLauC) [inline src/sys/socket/sockopt.rs:908]:
```suggestion
#[cfg(any(linux_android, bsd))]
```

DragonFlyBSD also has this, so `bsd` would work here

**Review by SteveLauC — CHANGES_REQUESTED** [review]

**Comment by SteveLauC** [comment]
BTW, a changelog is needed, please see [CONTRIBUTING.md](https://github.com/nix-rust/nix/blob/master/CONTRIBUTING.md#pull-requests) on how to add one.

If you apply the changes requested by me, then it should be something similar to:

> filename: 2113.added.md

```md
Add socket option `IPV6_PKTINFO` for BSDs/Linux/Android, also `IPV6_RECVPKTINFO` for DragonFlyBSD
```

File: src/sys/socket/sockopt.rs (hunk 2 above)
Comment (ishanjain28) [reply src/sys/socket/sockopt.rs]: Thank you for the links! I have updated it

File: src/sys/socket/sockopt.rs (hunk 3 above)
Comment (ishanjain28) [reply src/sys/socket/sockopt.rs:912]: @stevelauc This code is for the `IPV6_RECVPKTINFO` flag, The documentation incorrectly mentioned `IPV6_PKTINFO`

Comment (SteveLauC) [reply src/sys/socket/sockopt.rs:912]: It seems to be copied from the Linux manual: https://man7.org/linux/man-pages/man7/ipv6.7.html

> IPV6_RECVPKTINFO (since Linux 2.6.14)
              Set delivery of the IPV6_PKTINFO control message on
              incoming datagrams.  Such control messages contain a
              struct in6_pktinfo, as per RFC 3542.  Allowed only for
              SOCK_DGRAM or SOCK_RAW sockets.  Argument is a pointer to
              a boolean value in an integer.

Comment (ishanjain28) [reply src/sys/socket/sockopt.rs:912]: Ah my bad, Reverting it now

**Review by SteveLauC — APPROVED** [review]
Thanks! :)
